great_expectations/validator/validator.py

```diff
@@ -685,6 +687,21 @@ def get_config_value(self, key):
         """Getter for config value"""
         return self._validator_config.get(key)
 
+    def load_batch_from_batch_request(self, datasource, batch_request):
+        batches = datasource.get_batch_list_from_batch_request(batch_request)
+
+        for batch in batches:
+            self._execution_engine.load_batch_data(batch.id, batch.data)
+            self._batches[batch.id] = batch
+            self.active_batch_id = batch.id
```

Comment:
@talagluck Are you sure about this line?  It seems to change the value of the `active_batch_id` field on every iteration, so at the end, it would be that of the last batch's ID...

Reply:
Hi Alex! Yes - this is process that the `execution_engine` goes through in `load_batch_data`, and so I wanted to parallel it as closely as possible so the `active_batch` of the `Validator` and its `execution_engine` remain in sync.

Reply:
That makes sense to me. If that is the case, this is a very small optional suggestion that you could set the active_batch_id outside of the loop (so it is not set for each iteration), and maybe add a comment.
```suggestion
        # active_batch_id set to the last batch loaded to mirror `execution_engine` behavior
        self.active_batch_id = batch.id
```

Reply:
Sounds good - I think this might just be stylistic, and is something I want to think about. 

Realistically, if the loop gets interrupted, things will error and need to re-run, but I liked the idea of keeping the active_batch_id on the execution_engine and the validator the same throughout the process, even if it is only very briefly for the batches that aren't last. 

Comments are a

Reply:
Oh good point about the loop possibly getting interrupted, I take back my suggestion about un-indenting (but the comment might still be helpful)

Reply:
Yes! Comments are a great idea (is what I had started to type, but failed to finish above. (And have now edited to remove))

Reply:
The problem is that the line `self.active_batch_id = batch.id` is executed as part of every iteration of the loop.  So its effect is not clear.

Reply:
Why is this different from `self._execution_engine.load_batch_data(batch.id, batch.data)` being executed as part of every iteration of the loop?

Reply:
@talagluck Now I see what is going on!  Sorry that it took me so long to understand.  Here is the issue: I believe that the purpose of this code is to set the active_batch_id to that of the last batch in the list.  If that is the case, then the implementation should be more efficient and easier to interpret -- in both ExecutionEngine and elsewhere.  The mechanism should be more deliberate: instead of overwriting the value at each iteration of the loop, algorithmically determine the index of the last batch and set it accordingly -- this could be done in a couple of lines (or even one line).  This will be easier to understand and to modify, should the algorithm be changed.  /cc @anthonyburdi

Reply:
Hi @alexsherstinsky! Understood. I would say that the purpose of the code is not to set the active_batch_id to the last batch in the list, but to load additional batches to a validator. A side effect of this is that the active_batch_id of the validator should change to the last batch loaded. The active_batch_id of the execution_engine will change with every batch loaded, also as a side effect, and so I am trying to keep this consistent with that, so that we don't wind up with a situation where the active_batch on the execution_engine has been update, while the active_batch on the validator has not - it seems important to me that they remain in sync.

Reply:
@talagluck Agree -- I am saying the implementation for *both* must be changed, starting with the one in ExecutionEngine.

Reply:
@alexsherstinsky - see the point above re: interruption in the middle of the loop. I think it's fair to set each new batch loaded as the active one. That batch then becomes the cursor we discussed until a new one is loaded. In this case, the active_batch keeps track of the last batch loaded, and if something causes the process to interrupt, we have a record of progress.

Reply:
OK, @talagluck -- I see your point -- in case of an exception, we will at least have the current state.  Might you be interested in adding a comment and/or docstring in both places in order to prevent future such questions from causing long discussions?  Thank you!

Reply:
Hi @alexsherstinsky! I added a comment to `Validator`. Where else do you think would be a good place to document? The call stack is `data_context.get_batch_list` -> `datasource.get_batch_list_from_batch_request` -> `data_connector.get_batch_data_and_metadata` -> `execution_engine.load_batch_data`. The actual looping takes place in the `data_connector`.  I think that putting a note in the `execution_engine will just be confusing, since it's not immediately apparent that we are calling it on a loop there.